hashicorp/boundary-ui
# chore: 🤖 Migrate `form.input` to `Hds::TextInput::Field`

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12461
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12462
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12463
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12464
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12465

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12461)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12462)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12463)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12464)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12465)

## Description
- Migrate `form/credential` forms for for `json`, `ssh_private_key`, `username_password`, and `form/credential-store` forms for `static` and `vault` to use `Hds::TextInput::Field` instead of `form.input`.
- Fix corresponding test 
- Removed `Optional` helper text from `private_key_passphrase` translations because he can just use the `@optional` argument for the text field

## Screenshots (if appropriate):
- NOTE: To many screenshots to make this description long

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using yarn start
- Click on the "Orgs" in sidebar navigation and choose any `org`
- Click on any `project` card
- Click on "Credential Stores" in sidebar navigation
    - FOR TESTING CREDENTIAL STORE
        - Click on "New Credential Store" and ensure that when you click on a type card that all input fields are represented and inputed data is able to be saved.
    - FOR TESTING CREDENTIALS
        - Click on any `static` type store and click on "Credentials" tab 
        - Create `username_password`, `ssh_private_key` and `JSON` for static
        - Click on any `vault` type store and click "Manage" and "New Credential Library"
        - Ensure all  input fields are represented and inputed data is able to be saved.

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:


~~- [ ] I have added before and after screenshots for UI changes~~
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
